x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdate.java

```diff
@@ -193,7 +192,9 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalLong(modelSnapshotRetentionDays);
         out.writeOptionalLong(dailyModelSnapshotRetentionAfterDays);
         out.writeOptionalLong(resultsRetentionDays);
-        out.writeOptionalLong(systemAnnotationsRetentionDays);
+        if (out.getVersion().onOrAfter(Version.V_7_15_0) && out.getVersion().before(Version.V_8_0_0)) {
+            out.writeOptionalLong(null);
```

Comment:
I think *technically* this should be `out.writeOptionalLong(resultsRetentionDays);`

But, that doesn't matter really as when this change is backported this branch is removed.